pkg/checksum/checker.go

```diff
@@ -273,42 +264,45 @@ func (c *Checker) Run(ctx context.Context) error {
 	// Lock the source table in a trx
 	// so the connection is not used by others
 	c.logger.Info("starting checksum operation, this will require a table lock")
-	serverLock, err := dbconn.NewTableLock(ctx, c.db, c.table, false, c.dbConfig, c.logger)
+	serverLock, err := dbconn.NewTableLock(ctx, c.db, c.table, true, c.dbConfig, c.logger)
 	if err != nil {
 		return err
 	}
 	defer serverLock.Close()
-
 	// With the lock held, flush one more time under the lock tables.
 	// Because we know canal is up to date this now guarantees
 	// we have everything in the new table.
-	if err := c.feed.Flush(ctx); err != nil {
+	if err := c.feed.FlushUnderLock(ctx, serverLock); err != nil {
```

Comment:
IIUC, This is also a important change to reduce the chances the below check `AllChangesFlushed()` failing due to parallelism, but I guess not the reason the migrations under consideration failed, as they failed in the checksum phase.